File: Models/Seed.swift
import SwiftData

@Model
final class Seed {
    var id: UUID
    var cropName: String
    var variety: String?
    var state: SeedState
    var notes: String?
    var createdAt: Date
    var updatedAt: Date

    @Relationship(deleteRule: .nullify, inverse: \Garden.seeds)
    var garden: Garden?

    @Relationship(deleteRule: .nullify, inverse: \PlannedPlanting.seed)
    var plannedPlantings: [PlannedPlanting] = []

    init(cropName: String, variety: String? = nil, state: SeedState = .own, notes: String? = nil) {
        self.id = UUID()
        self.cropName = cropName
        self.variety = variety
        self.state = state
        self.notes = notes
        self.createdAt = Date()
        self.updatedAt = Date()
    }

    var displayName: String {
        if let variety = variety, !variety.isEmpty {
            return "\(variety) \(cropName)"
        }
        return cropName
    }
}

enum SeedState: String, CaseIterable {
    case own
    case want

    var displayName: String {
        switch self {
        case .own: return "I Have"
        case .want: return "I Want"
        }
    }

    var symbolName: String {
        switch self {
        case .own: return "checkmark.circle.fill"
        case .want: return "plus.circle.fill"
        }
    }
}

File: shared/src/commonMain/kotlin/com/soilandsupper/shared/domain/model/Seed.kt
package com.soilandsupper.shared.domain.model

data class Seed(
    val id: Long = 0,
    val cropName: String,
    val variety: String? = null,
    val state: String = SeedState.OWN.name,
    val notes: String? = null,
    val gardenId: Long? = null,
    val createdAt: Long = System.currentTimeMillis(),
    val updatedAt: Long = System.currentTimeMillis()
) {
    val displayName: String
        get() = if (!variety.isNullOrBlank()) "$variety $cropName" else cropName
}

